### PR TITLE
adding support for configmap namespace

### PR DESCRIPTION
**Pull request by djkormo**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Removing hardcoded namespace in mertics-exporter go code.
Adding proper Dockerfile for image.
Adding go modules to code.

#### Which issue(s) this PR fixes:

Automatically closes linked issue when PR is merged.
https://github.com/cri-o/cri-o/issues/6607

#### Does this PR introduce a user-facing change?

No

### CODE REVIEW

**Comment by amarlearning** [comment]
@djkormo Please update the answer for `Does this PR introduce a user-facing change?` in your PR description.

**Comment by saschagrunert** [comment]
@djkormo please sign-off your commits via `git commit -s --amend`

File: contrib/metrics-exporter/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+FROM golang:1.19 as builder
```
Comment (saschagrunert) [inline contrib/metrics-exporter/Dockerfile:2]: Can we add this to dependencies.yaml:

https://github.com/cri-o/cri-o/blob/6e861a881c1c7124836228c7c7b9abb696f6c283/dependencies.yaml#L1-L14

File: contrib/metrics-exporter/main.go [hunk 2]
```diff
@@ -17,12 +17,9 @@ import (
...
 
-const (
-	namespace = "cri-o-metrics-exporter"
-	service   = namespace
-	configMap = namespace
-)
-
+var namespace = env.Default("POD_NAMESPACE", "cri-o-metrics-exporter")
+var service = namespace  
+var configMap = namespace
```
Comment (saschagrunert) [inline contrib/metrics-exporter/main.go:22]:
```suggestion
var (
	namespace = env.Default("POD_NAMESPACE", "cri-o-metrics-exporter")
	service = namespace  
	configMap = namespace
)
```

File: contrib/metrics-exporter/Dockerfile (hunk 1 above)
Comment (djkormo) [reply contrib/metrics-exporter/Dockerfile:2]: Done!

File: contrib/metrics-exporter/main.go (hunk 2 above)
Comment (djkormo) [reply contrib/metrics-exporter/main.go]: Done!

**Comment by haircommander** [comment]
/ok-to-test
/approve

thanks @djkormo ! can you squash the two commits together please? otherwise, LGTM

File: contrib/metrics-exporter/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+
+
+Proper Dockerfile for application
+
+```Dockerfile
```
Comment (haircommander) [inline contrib/metrics-exporter/README.md:5]: this seems prone to skewing from the Dockerfile in the same directory. Can we link instead?

File: contrib/metrics-exporter/main.go [hunk 4]
```diff
@@ -17,12 +17,11 @@ import (
 	"sigs.k8s.io/release-utils/env"
 )
 
-const (
-	namespace = "cri-o-metrics-exporter"
-	service   = namespace
+var (
+	namespace = env.Default("POD_NAMESPACE", "cri-o-metrics-exporter")
+	service = namespace
```
Comment (haircommander) [inline contrib/metrics-exporter/main.go:22]: hm there seems to be some whitespace weirdness here

**Comment by sohankunkerkar** [comment]
@djkormo do you have the bandwidth to address Peter's comments and get this PR to the finish line?

**Comment by amarlearning** [comment]
@djkormo @sohankunkerkar I can go ahead and make appropriate changes to close this PR ASAP.

**Comment by amarlearning** [comment]
@sohankunkerkar @haircommander It seems like @djkormo is not getting time to finish to this PR, shall I go ahead?

**Comment by haircommander** [comment]
> @sohankunkerkar @haircommander It seems like @djkormo is not getting time to finish to this PR, shall I go ahead?

sure @amarlearning thanks!

**Comment by amarlearning** [comment]
@haircommander update: stuck with office work, will try to close the comments in coming weekend.

**Comment by amarlearning** [comment]
@haircommander @saschagrunert @sohankunkerkar I have requested changes but couldn't push them to the @djkormo clone. Shall I make the necessary changes in my clone and create a new PR?

**Comment by amarlearning** [comment]
I went ahead and created a [new](https://github.com/cri-o/cri-o/pull/6891) PR with @djkormo commits.

**Pull request closed without merging**